src/Parser/Core/Modules/Items/SephuzsSecret.js

```diff
@@ -1,46 +1,39 @@
+import React from 'react';
+
 import SPELLS from 'common/SPELLS';
 import ITEMS from 'common/ITEMS';
 import { formatPercentage } from 'common/format';
 
 import Module from 'Parser/Core/Module';
 import Combatants from 'Parser/Core/Modules/Combatants';
 
+const PASSIVE_HASTE = 0.02;
+const ACTIVE_HASTE = 0.25;
+
 class SephuzsSecret extends Module {
   static dependencies = {
     combatants: Combatants,
   };
-  uptime = 0;
 
   on_initialized() {
     this.active = this.combatants.selected.hasFinger(ITEMS.SEPHUZS_SECRET.id);
   }
 
-  lastAppliedTimestamp = null;
-  on_toPlayer_applybuff(event) {
-    const spellId = event.ability.guid;
-
-    if (spellId === SPELLS.SEPHUZS_SECRET_BUFF.id) {
-      this.lastAppliedTimestamp = event.timestamp;
-    }
-  }
-  on_toPlayer_removebuff(event) {
-    const spellId = event.ability.guid;
-
-    if (spellId === SPELLS.SEPHUZS_SECRET_BUFF.id) {
-      this.uptime += event.timestamp - this.lastAppliedTimestamp;
-      this.lastAppliedTimestamp = null;
-    }
-  }
-  on_finished() {
-    if (this.lastAppliedTimestamp) {
-      this.uptime += this.owner.fight.end_time - this.lastAppliedTimestamp;
-    }
-  }
-
   item() {
+    const uptimePercent = this.combatants.selected.getBuffUptime(SPELLS.SEPHUZS_SECRET_BUFF.id) / this.owner.fightDuration;
+    const avgHaste = (uptimePercent * ACTIVE_HASTE) + ((1 - uptimePercent) * PASSIVE_HASTE);
+
     return {
       item: ITEMS.SEPHUZS_SECRET,
-      result: `${formatPercentage((this.uptime / this.owner.fightDuration) || 0)} % uptime`,
+      result: (
+        <span>
+          <dfn
+            data-tip={`This is the average haste percentage gained, factoring in both the passive and active bonuses. The active's uptime was <b>${formatPercentage(uptimePercent)}%</b>`}
+          >
+            {formatPercentage(avgHaste)} % average haste
```

Comment:
I'd much rather have this show haste rating (or both % and rating), it's easier to use to compare stuff.

Reply:
Haste rating isn't correct though, as Sephuz buff stacks multiplicatively with rating haste